Using entity_by_index and correcting entries in surface map.

pshriwise opened the pull request:
This PR updates uses `entity_by_index` lookups for volume handles in the DAGMC interface as they perform much better than the current `entity_by_id` lookups. I was going to use the `surface/cell_map`s to find these indices but I decided on adding an attribute to the `DAGMCSurface/Cell` classes to contain this 1-indexing to DAGMC-related code. @paulromano did just go through all the effort of removing those +1's after all...

I also replaced the direction values in `plot.cpp` with a normalized direction which was resulting in false negatives in DAGMC's `point_in_volume` method due to problems when trying to intersect that direction with triangles in the surface mesh.

Overall effects of this PR are significant improvement a significant improvement in tracking and plotting performance for CAD-based geometries and a fix to resulting images when plotting CAD-based geometries as well.

makeclean commented on src/dagmc.cpp:
```diff
@@ -303,7 +303,7 @@ void load_dagmc_geometry()
     // add to global array and map
     model::surfaces.emplace_back(s);
-    model::surface_map[s->id_] = s->id_;
+    model::surface_map[s->id_] = i;
```
presumably ok to start at 0?

makeclean replied:
Might be being picky, but i is an awful long way from its definition, maybe this function should be broken up or something? Not really sure what would make it better?

pshriwise commented:
Ok well now that I've added a few missing commits (didn't catch a needed force push after rebasing), I've also moved the use of those iterator values closer to the for loop statement for better clarity. Is that ok for now @makeclean?

makeclean commented:
LGTM - Can I hit merge?

paulromano commented:
@makeclean Yes, you may do the honors. Just so you're aware, we have a mandatory 36 hour review period (described [here](https://openmc.readthedocs.io/en/latest/devguide/contributing.html#contribution-process)) for non-trivial PRs. That being said, this one would probably fall in the trivial category (open to the interpretation of the reviewers).

makeclean commented:
Woo!
